patchwork-level-generators/src/main/java/net/patchworkmc/mixin/levelgenerators/MixinBiomeLayers.java

```diff
@@ -0,0 +1,133 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+ 
+package net.patchworkmc.mixin.levelgenerators;
+
+import java.util.function.LongFunction;
+
+import com.google.common.collect.ImmutableList;
+import net.minecraftforge.common.extensions.IForgeWorldType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.world.biome.layer.AddClimateLayers;
+import net.minecraft.world.biome.layer.AddColdClimatesLayer;
+import net.minecraft.world.biome.layer.AddDeepOceanLayer;
+import net.minecraft.world.biome.layer.AddEdgeBiomesLayer;
+import net.minecraft.world.biome.layer.AddHillsLayer;
+import net.minecraft.world.biome.layer.AddIslandLayer;
+import net.minecraft.world.biome.layer.AddMushroomIslandLayer;
+import net.minecraft.world.biome.layer.AddRiversLayer;
+import net.minecraft.world.biome.layer.AddSunflowerPlainsLayer;
+import net.minecraft.world.biome.layer.ApplyOceanTemperatureLayer;
+import net.minecraft.world.biome.layer.BiomeLayers;
+import net.minecraft.world.biome.layer.CellScaleLayer;
+import net.minecraft.world.biome.layer.ContinentLayer;
+import net.minecraft.world.biome.layer.IncreaseEdgeCurvatureLayer;
+import net.minecraft.world.biome.layer.NoiseToRiverLayer;
+import net.minecraft.world.biome.layer.OceanTemperatureLayer;
+import net.minecraft.world.biome.layer.ScaleLayer;
+import net.minecraft.world.biome.layer.SimpleLandNoiseLayer;
+import net.minecraft.world.biome.layer.SmoothenShorelineLayer;
+import net.minecraft.world.biome.layer.type.ParentedLayer;
+import net.minecraft.world.biome.layer.util.LayerFactory;
+import net.minecraft.world.biome.layer.util.LayerSampleContext;
+import net.minecraft.world.biome.layer.util.LayerSampler;
+import net.minecraft.world.gen.chunk.OverworldChunkGeneratorConfig;
+import net.minecraft.world.level.LevelGeneratorType;
+
+import net.patchworkmc.impl.levelgenerators.PatchworkGeneratorType;
+
+@Mixin(BiomeLayers.class)
+public class MixinBiomeLayers {
+	@Shadow
+	private static <T extends LayerSampler, C extends LayerSampleContext<T>> LayerFactory<T> stack(long seed, ParentedLayer layer, LayerFactory<T> parent, int count, LongFunction<C> contextProvider) {
+		throw new RuntimeException("Failed to shadow BiomeLayers#stack!");
+	}
+
+	@Inject(at = @At("HEAD"), method = "build", cancellable = true)
+	private static <T extends LayerSampler, C extends LayerSampleContext<T>> void build(LevelGeneratorType generatorType, OverworldChunkGeneratorConfig settings, LongFunction<C> contextProvider, CallbackInfoReturnable<ImmutableList<LayerFactory<T>>> info) {
+		if (generatorType instanceof PatchworkGeneratorType) {
+			LayerFactory<T> continentLayer = ContinentLayer.INSTANCE.create(contextProvider.apply(1L));
```

Comment:
Woah, could we get some spacing between blocks here?

Reply:
checkstyle complains

Reply:
oh wait